### ISSUE TITLE
Use zope.interface class decorators in twisted/test

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5843]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)| @vperic reported|
|-|-|
|Trac ID|trac#5843|
|Type|task|
|Created|2012-08-04 09:32:30Z|

After [#5683](https://github.com/twisted/twisted/issues/5683) gets in, the uses of implements() should be changed to the new class decorator, @implementer, for Python 3 compatibility. 

Attachments:

* [implementer-test.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/356/356b27f7b843c11e49e684f0e5c9c8179331ca79/20671f146be4d8072ba37649efb5370e21ecf9e1.patch) (19181 bytes) - added by vperic on 2012-08-04 09:32:50Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5843 5843
type__task task
reporter__vperic vperic
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__core core
keywords__ 
time__1344072750000000 1344072750000000
changetime__1349886136000000 1349886136000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_1"></a>|@vperic commented|
|-|-|

Well, [#5863](https://github.com/twisted/twisted/issues/5863) is in, so this should get looked at. It's a fairly hefty patch, but it's in twisted.test so at least we know it's all tested. I'll probably put this on a branch ASAP so we can notice which builders fail (don't have the required version of zope.interface).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

This change should not be merged until 12.2 is released, also see [ticket:5683#comment:22 this comment].

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

Thanks for the patch, but we're modifying module one by one, instead of doing a bulk change.
